From 6d3c4a9963c77d6c98f5d495f62b6605428f4fff Mon Sep 17 00:00:00 2001
From: Xinyu Chen <xinyu.chen@freescale.com>
Date: Wed, 18 Jul 2012 10:22:26 +0800
Subject: [PATCH 310/498] ENGR00217401-2 build warning: sensor name should be
 const char

Each sensor name should be assigned to const char

Signed-off-by: Xinyu Chen <xinyu.chen@freescale.com>
---
 libsensors/MagSensor.cpp   | 4 ++--
 libsensors/PressSensor.cpp | 2 +-
 libsensors/SensorBase.cpp  | 2 +-
 3 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/libsensors/MagSensor.cpp b/libsensors/MagSensor.cpp
index 1090fd3..bdf51ce 100755
--- a/libsensors/MagSensor.cpp
+++ b/libsensors/MagSensor.cpp
@@ -37,8 +37,8 @@
 MagSensor::MagSensor()
 : SensorBase(NULL, "eCompass")
 {
-    char * magSensorName = "mag3110";
-	sensorBaseGetSysfsPath(magSensorName);
+    const char *magSensorName = "mag3110";
+    sensorBaseGetSysfsPath(magSensorName);
     memset(&mPendingEvents[MagneticField], 0, sizeof(sensors_event_t));
     memset(&mPendingEvents[Orientation], 0, sizeof(sensors_event_t));
     mPendingEvents[MagneticField].version = sizeof(sensors_event_t);
diff --git a/libsensors/PressSensor.cpp b/libsensors/PressSensor.cpp
index 5caec07..ab83461 100755
--- a/libsensors/PressSensor.cpp
+++ b/libsensors/PressSensor.cpp
@@ -31,7 +31,7 @@
 PressSensor::PressSensor()
 :SensorBase(NULL, "mpl3115")
 {
-    char * magSensorName = "mpl3115";
+    const char *magSensorName = "mpl3115";
     sensorBaseGetSysfsPath(magSensorName);
     memset(&mPendingEvents[Pressure], 0, sizeof(sensors_event_t));
     memset(&mPendingEvents[Temperatury], 0, sizeof(sensors_event_t));
diff --git a/libsensors/SensorBase.cpp b/libsensors/SensorBase.cpp
index 5aba33e..07d3415 100755
--- a/libsensors/SensorBase.cpp
+++ b/libsensors/SensorBase.cpp
@@ -124,7 +124,7 @@ int SensorBase::sensorBaseEnable(int32_t handle,int enabled){
         case ID_PX 	: what = Proximity;   break;
 
     }
-    if (uint32_t(what) >= numSensors)
+    if (what < 0 || what >= numSensors)
         return -EINVAL;
 
     if(enable)
-- 
1.8.0

